syntax = "proto3";

package protoships;


service ShipsService {
    rpc GetAircraft(RequestSpaceCoor) returns (stream ResponceAircraft);
}

message RequestSpaceCoor {
   repeated float coordinates = 1;
}

enum AlignmentType{
    AlignmentUnknown = 0;
    Ally = 1;
    Enemy = 2;
}

enum ClassEnum{
    ClassUnknown = 0;
    Corvette = 1;
    Frigate = 2;
    Cruiser = 3;
    Destroyer = 4;
    Carrier = 5;
    Dreadnought = 6;
}

message ResponceAircraft {
    AlignmentType aligment = 1;
    string name = 2;
    ClassEnum _class = 3;
    float length = 4;
    int32 crew_size = 5;
    bool armed = 6;
    message ListItems {
        string first_name = 1;
        string last_name = 2;
        string rank = 3;
    }
    repeated ListItems officers = 7;
}
